src/Status.hpp

```diff
@@ -0,0 +1,155 @@
+#pragma once
+
+#include <atomic>
+#include <chrono>
+#include <cmath>
+#include <iostream>
+#include <memory>
+#include <mutex>
+#include <vector>
+
+#include "utils.h"
+
+namespace FileWriter {
+class Streamer;
```

Comment:
Streamer is originally defined as `struct`, forward declaring as class causes a warning.

Reply:
:white_check_mark: